src/backends/getrandom.rs

```diff
@@ -26,6 +26,14 @@ mod util_libc;
 #[inline]
 pub fn fill_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
     util_libc::sys_fill_exact(dest, |buf| unsafe {
-        libc::getrandom(buf.as_mut_ptr().cast(), buf.len(), 0)
+        let ret = libc::getrandom(buf.as_mut_ptr().cast(), buf.len(), 0);
+
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        #[allow(unused_unsafe)] // TODO(MSRV 1.65): Remove this.
+        unsafe {
+            super::sanitizer::unpoison_linux_getrandom_result(buf, ret);
```

Comment:
Is unpoisoning needed if we are calling the `libc` version of `getrandom`? The comment in `sanitizer.rs` seems to indicate that the `libc` calls should handle unpoisioning.

Reply:
See https://github.com/rust-random/getrandom/blob/5f7a91020fb75435dfe063a0c231010c9163f32a/src/backends/sanitizer.rs#L42-L48. Basically the way MSAN is configured by Rust, to support older glibc, ensures that the interception is disabled for glibc. And, because of how MSAN won't do interception for non-glibc.

It is possible that the libc has interception built-in, but we have no way of knowing. I think common open source implementations like musl libc don't, except maybe LLVM libc? If we have a way to detect that LLVM libc will be used, we could make an exception for it.

Reply:
I was confused mostly because `libc::getrandom` will only exist on GLibc 2.25 or later, so I figured we would either have:
  - An older glibc, so we fall back to `/dev/urandom` or fail depending on the backend
  - A glibc >= 2.25, so `libc::getrandom` exists and get proprerly handled with msan/libc

But if there's an extra level of config between Rust libc and msan, it totally makes sense why this wouldn't work.

Reply:
@josephlr Do you know what the Android bionic situation is? IDK if MSAN even works for Android. Would love to minimize the use of this as much as possible.

Reply:
I have personally not ever used bionic with MSAN. When fuzzing or running tests against C libraries, we usually just run them on Linux, as most of the validation is not for OS specific stuff. But that doesn't mean it's not possible, I've asked some folks at work and we'll see if they get back to me.

Reply:
I couldnt find any mention about MSan in Android's memory debugging guide: https://developer.android.com/ndk/guides/memory-debug and all the other tools I found were also of an ASan like flavor.

I would be fine assuming that:
  1) Android Bionic (as part of the NDK) doesn't support MSan
  2) if this support is added, it will correctly handle the libc-based functions which write buffers (`libc::read`, `libc::getrandom`, etc...) unconditionally, as newer versions of the libc will always have get random

I might be wrong, but it seems fine to remove this for android.